libeos-payg/manager.c

```diff
@@ -1336,29 +1336,17 @@ static void file_save_delete_cb (GObject      *source_object,
                                  GAsyncResult *result,
                                  gpointer      user_data);
 static void
-write_guint64_to_file (guint64       number,
-                       GFile        *file,
-                       GTask        *task,
-                       GCancellable *cancellable)
+write_guint64_to_file (guint64  number,
+                       GFile   *file,
+                       GTask   *task)
 {
-  union
-    {
-      guint64 u64;
-      const guint8 u8[8];
-    } number_union;
-  number_union.u64 = number;
-
-  g_autoptr(GBytes) number_bytes = g_bytes_new (number_union.u8,
-                                                sizeof (number_union.u8));
-
-  g_file_replace_contents_bytes_async (file,
-                                       number_bytes,
-                                       NULL,  /* ETag */
-                                       FALSE,  /* no backup */
-                                       G_FILE_CREATE_PRIVATE,
-                                       cancellable,
-                                       file_replace_cb,
-                                       g_object_ref (task));
+  g_autofree gchar *file_path = g_file_get_path (file);
+  g_autoptr(GError) local_error = NULL;
+
+  if (!g_file_set_contents (file_path, (char *)&number, sizeof(number), &local_error))
```

Comment:
Not to be That Guy again, but this is synchronous and will block the main thread.